### DISCUSSION TITLE
Denoiser implementation?

### DISCUSSION THREAD

**Discussion by aVersionOfReality** [bnpr/Malt#141] [Q&A] [answered]
Would it be possible to implement a denoiser, like those currently used by Cycles, into Malt? I assume it could be done in a screen shader? What sort of thing would be involved? Or is this just impossible?

**Comment by pragma37** [comment, marked as answer]
Sure it's possible. What's the use-case, though?

There aren't many features that can actually generate noise, and rendering more samples is usually cheap enough.
For edge cases, you could always use the denoiser nodes from the Blender compositor.

**Reply by aVersionOfReality** [reply]
Noise is pretty common when using large amounts of Normal Beveling. Even huge amounts of samples don't clean it up. I'm also looking into ways to address issues in toon shading that are caused by Smooth Shading, and that may involve blurring of some sort too.
Mostly I'm just trying to get an idea of what is already implemented, what can be implemented, and what can never be implemented.
